frontends/src/frontend.rs

```diff
@@ -15,34 +15,18 @@ pub enum Event {
 
 /// A front end for the editor. It knows how to render to a screen,
 /// and how to receive keyboard events.
-pub trait Frontend: Sized + PrettyScreen {
+pub trait Frontend: Sized {
+    type Error;
```

Comment:
Done. The `Frontend` trait now has an associated type `Window: PrettyWindow`, and the closure takes a `Pane<Self::Window>`. This allows flexibility in whether the `Terminal` struct will implement `PrettyWindow` itself, or a `TerminalWrapper<'a>(&'a mut Terminal)` will implement `PrettyWindow` instead. I'm sticking with having `Terminal` implement `PrettyWindow` for now, because the wrapper doesn't seem valuable enough to justify another descent into the hell of 'lifetime annotations in traits'.